shared/actions/settings.js

```diff
@@ -1,18 +1,70 @@
 // @flow
 import * as Constants from '../constants/settings'
-import {apiserverGetRpcPromise} from '../constants/types/flow-types'
-import {call, put} from 'redux-saga/effects'
+import {apiserverGetRpcPromise, apiserverPostJSONRpcPromise} from '../constants/types/flow-types'
+import {call, put, select} from 'redux-saga/effects'
 import {takeLatest} from 'redux-saga'
 
 import type {SagaGenerator} from '../constants/types/saga'
-import type {NotificationsRefresh} from '../constants/settings'
+import type {NotificationsRefresh, NotificationsSave, NotificationsToggle} from '../constants/settings'
 
 function notificationsRefresh (): NotificationsRefresh {
   return {type: Constants.notificationsRefresh, payload: undefined}
 }
 
+function notificationsSave (): NotificationsSave {
+  return {type: Constants.notificationsSave, payload: undefined}
+}
+
+function notificationsToggle (name?: string): NotificationsToggle {
+  return {type: Constants.notificationsToggle, payload: {name}}
+}
+
+function * saveNotificationsSaga (): SagaGenerator<any, any> {
+  try {
+    const current = yield select(state => state.settings.notifications)
+
+    if (!current || !current.settings) {
+      throw new Error('No notifications loaded yet')
+    }
+
+    const JSONPayload = current.settings
+      .map(s => ({
+        key: `${s.name}|email`,
+        value: s.subscribed ? '1' : '0'}))
+      .concat({
+        key: `unsub|email`,
+        value: current.unsubscribedFromAll ? '1' : '0'})
+
+    const result = yield call(apiserverPostJSONRpcPromise, {
+      param: {
+        endpoint: 'account/subscribe',
+        args: [],
+        JSONPayload,
+      },
+    })
+
+    if (JSON.parse(result && result.body || '').status.code !== 0) {
+      throw new Error(`Invalid response ${result || ''}`)
+    }
+
+    yield put({
+      type: Constants.notificationsSaved,
+      payload: undefined,
+    })
+  } catch (err) {
+    // TODO global error
+  }
+}
+
 function * refreshNotificationsSaga (): SagaGenerator<any, any> {
   try {
+    yield put({
```

Comment:
We should probably have a reusable helper that will set a loading status after N ms of working -- perhaps we already have something close to fitting the bill?